src/WebJobs.Script.WebHost/Controllers/AdminController.cs

```diff
@@ -86,153 +90,139 @@ public FunctionStatus GetFunctionStatus(string name)
+        [Authorize(Policy = PolicyNames.AdminAuthLevelOrInternal)]
+        public IActionResult GetHostStatus()
         {
-            var authorizationLevel = Request.GetAuthorizationLevel();
-            if (Request.IsAuthDisabled() ||
-                authorizationLevel == AuthorizationLevel.Admin ||
-                Request.IsAntaresInternalRequest())
+            var status = new HostStatus
             {
-                var status = new HostStatus
-                {
-                    State = _scriptHostManager.State.ToString(),
-                    Version = ScriptHost.Version,
-                    Id = _scriptHostManager.Instance?.ScriptConfig.HostConfig.HostId
-                };
+                State = _scriptHostManager.State.ToString(),
+                Version = ScriptHost.Version,
+                VersionDetails = Utility.GetInformationalVersion(typeof(ScriptHost)),
+                Id = _scriptHostManager.Instance?.ScriptConfig.HostConfig.HostId
+            };
 
-                var lastError = _scriptHostManager.LastError;
-                if (lastError != null)
-                {
-                    status.Errors = new Collection<string>();
-                    status.Errors.Add(Utility.FlattenException(lastError));
-                }
+            var lastError = _scriptHostManager.LastError;
+            if (lastError != null)
+            {
+                status.Errors = new Collection<string>();
+                status.Errors.Add(Utility.FlattenException(lastError));
+            }
 
-                var parameters = Request.GetQueryParameterDictionary();
-                string value = null;
-                if (parameters.TryGetValue(ScriptConstants.CheckLoadQueryParameterName, out value) && value == "1")
+            var parameters = Request.Query;
+            if (parameters.TryGetValue(ScriptConstants.CheckLoadQueryParameterName, out StringValues value) && value == "1")
+            {
+                status.Load = new LoadStatus
                 {
-                    status.Load = new LoadStatus
-                    {
-                        IsHigh = _scriptHostManager.PerformanceManager.IsUnderHighLoad()
-                    };
-                }
+                    IsHigh = _scriptHostManager.PerformanceManager.IsUnderHighLoad()
+                };
+            }
 
-                string message = $"Host Status: {JsonConvert.SerializeObject(status, Formatting.Indented)}";
-                _traceWriter.Info(message);
-                _logger?.LogInformation(message);
+            string message = $"Host Status: {JsonConvert.SerializeObject(status, Formatting.Indented)}";
+            _logger?.LogInformation(message);
 
-                return Ok(status);
-            }
-            else
-            {
-                return Unauthorized();
-            }
+            return Ok(status);
         }
 
         [HttpPost]
         [Route("admin/host/ping")]
         [AllowAnonymous]
```

Comment:
AllowAnonymous not needed anymore if you've removed class level auth?

Reply:
Correct. Removed.